daemon/delete.go

```diff
@@ -76,22 +76,20 @@ func (daemon *Daemon) rm(container *Container, forceRemove bool) (err error) {
 		}
 	}
 
-	// stop collection of stats for the container regardless
-	// if stats are currently getting collected.
-	daemon.statsCollector.stopCollection(container)
-
-	element := daemon.containers.Get(container.ID)
```

Comment:
this check was inherited from when the delete method was not thread safe, long long time ago :scream: